### PR TITLE
HIVE-27416: ClassCastException for string to decimal64 conversion

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Added string to decimal64 converter

### Why are the changes needed?

ClassCastException

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

q test

### CODE REVIEW

**Comment by aturoczy** [comment]
Sorry for the dumb question, but the BigDecimal is not good for this? As I remember the BigDecimal and BigInteger has a string constructor and could receives strings. Maybe I misunderstood this code (most likely) just I rather ask stupid question if I something bothering me :)  I guess it has an intention that I'm not aware or just not understand correctly.

**Comment by aturoczy** [comment]
Nothing, just realized this is Decimal64ColumnVector.  It is not pure BigDecimal.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/CastStringToDecimal.java [hunk 1]
```diff
@@ -66,6 +67,10 @@ protected void func(DecimalColumnVector outputColVector, BytesColumnVector input
     }
   }
 
+  protected void setOutputColumnVectorValue(ColumnVector outputColVector, int i, String s) {
+    ((DecimalColumnVector) outputColVector).set(i, HiveDecimal.create(s));
```
[the remark this answers is not in the document]
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/CastStringToDecimal.java:71]: `set` method is different for DecimalColumnVector and Decimal64ColumnVector

[the remark this answers is not in the document]
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/CastStringToDecimal.java]: added `void IDecimalColumnVector.set(int elementNum, HiveDecimal decimal)` to `IDecimalColumnVector` interface

**Review by aturoczy — APPROVED** [review]

**Pull request merged**